src/main/java/net/minecraftforge/client/gui/config/widgets/ConfigGuiScreen.java

```diff
@@ -0,0 +1,723 @@
+    private static final ResourceLocation FORGE_SEARCH_ICON = new ResourceLocation("forge", "textures/gui/search.png");
+    private final Map<List<String>, Object> currentValues = Maps.newHashMap();
+    private final Map<List<String>, Object> initialValues = Maps.newHashMap();
+    private final Collection<SpecificationData> specs;
+    private final Set<ConfigEntry> invalidEntries = Sets.newHashSet();
+    private final Runnable onClose;
+    private ConfigGuiScreen.ConfigEntryList configEntryList;
+    private Button doneButton;
+    private ImageContentButton resetToInitialButton = null;
+    private ImageContentButton resetToDefaultButton = null;
+    @Nullable
+    private List<FormattedCharSequence> tooltip;
+
+    public ConfigGuiScreen(Component title, final Collection<SpecificationData> specs, Runnable onClose)
+    {
+        super(title);
+        this.specs = specs;
+        this.onClose = onClose;
+    }
+
+    @NotNull
+    private static Comparator<List<String>> createListComparator()
+    {
+        return (o1, o2) ->
+        {
+            if (o1.size() == o2.size())
+            {
+                for (int i = 0; i < o1.size(); i++)
+                {
+                    final int result = o1.get(i).compareTo(o2.get(i));
+                    if (result != 0)
+                    {
+                        return result;
+                    }
+                }
+                return 0;
+            }
+
+            return o1.size() - o2.size();
+        };
+    }
+
+    @Override
+    public void onClose()
+    {
+        super.onClose();
+        this.onClose.run();
+    }
+
+    @Override
+    public boolean keyPressed(final int key, final int scanCode, final int modifiers)
+    {
+        if (key == 256 && this.shouldCloseOnEsc())
+        {
+            this.configEntryList.onCancel(this::onClose);
+            return true;
+        }
+
+        return super.keyPressed(key, scanCode, modifiers);
+    }
+
+    @Override
+    protected void init()
+    {
+        Objects.requireNonNull(this.minecraft).keyboardHandler.setSendRepeatsToGui(true);
+        super.init();
+        this.configEntryList = new ConfigEntryList();
+        this.addWidget(this.configEntryList);
+        this.addRenderableWidget(new Button(this.width / 2 - 155 + 160, this.height - 29, 150, 20,
+                CommonComponents.GUI_CANCEL, (button) -> this.configEntryList.onCancel(this::onClose)));
+        this.doneButton = this.addRenderableWidget(new Button(this.width / 2 - 155, this.height - 29, 150, 20,
+                CommonComponents.GUI_DONE, (button) -> {
+            this.configEntryList.onSave();
+            this.onClose();
+        }));
+        final EditBox searchBox = this.addRenderableWidget(new EditBox(this.font, this.width / 2 + 32, 16,
+                (this.width - 100) / 2, 20, Component.translatable("forge.configgui.search")));
```

Comment:
Sadly I tried that but that breaks the scroll list.
Yes this will reset the search but it also resets the entire list this way which is okeyish.

Mojangs implementation of the scroll list is crap and it does not really like doing fancy things to it. So for right now I will leave it like this.